widgets/text/text_test.go

```diff
@@ -809,6 +809,112 @@ func TestTextDraws(t *testing.T) {
 				return ft
 			},
 		},
+		{
+			desc:   "tests maxContent length being applied - multiline",
+			canvas: image.Rect(0, 0, 10, 3),
+			opts: []Option{
+				MaxContent(10),
+				RollContent(),
+			},
+			writes: func(widget *Text) error {
+				return widget.Write("line0\nline1\nline2\nline3\nline4")
+			},
+			want: func(size image.Point) *faketerm.Terminal {
+				ft := faketerm.MustNew(size)
+				c := testcanvas.MustNew(ft.Area())
+				// \n still counts as a chacter in the string length
+				testdraw.MustText(c, "ine3", image.Point{0, 0})
+				testdraw.MustText(c, "line4", image.Point{0, 1})
+				testcanvas.MustApply(c, ft)
+				return ft
+			},
+		},
+		{
+			desc:   "tests maxContent exact length of 5",
```

Comment:
Other interesting test cases are:

- length is just one under the set max
- length is just one over the set max
- multiple writes, the last write causes the content to be reset because the provided text is longer than the configured max
- multiple writes, the last writes truncates the existing content before writing
- multiple writes, the last write truncates text but leaves content untouched

We can also take a look at the coverage report to see which lines aren't covered.